http/src/lib.rs

```diff
@@ -554,7 +554,7 @@ fn configure_port(_reuse: bool, _tcp: &net2::TcpBuilder) -> io::Result<()> {
 pub struct Server {
 	address: SocketAddr,
 	executor: Option<Vec<Executor>>,
```

Comment:
executor should most likely be wrapped in `Arc<Mutex<` as wel, so that it can be passed to `CloseHandle. Actually it would be simpler to store `CloseHandle` directly in `Server` and just clone it when requested. `CloseHandle` should handle the case if `Options` are `None`.
To simplify it further you can just have:
```rust

pub struct Server {
   address: SocketAddr,
  close_handle: CloseHandle,
}

impl Server {
  ...
  pub fn close_handle(&self) -> CloseHandle {
    self.close_handle.clone()
  }
}

#[derive(Clone)]
pub struct CloseHandle {
  handle: Arc<Mutex<Option<(Executor, Vec<oneshot::Sender<()>)>>>,
}

impl CloseHandle {
  ///  <docs ...>
  /// returns `false` if the server was closed earlier
  pub fn close(self) -> bool {
     if let Some((executor, senders)) = self.handle.lock().take() {
        executor.close();
        for s in senders { s.send(()).expect(...) } 
        true
     } else { false }
  }
}
```

Reply:
Not sure how this would work: once the server starts waiting it has to take the lock on the `handle`, so `close()` as written above will hang on the lock?